### DISCUSSION TITLE
Run RDKit single-threaded?

### DISCUSSION THREAD

**Discussion by apahl** [rdkit/rdkit#6784] [Q&A] [answered]
Hi,

we are using the RDKit to analyze structures on a compute cluster.  
We are getting different results for some cases when the data is run on the nodes in parallel (with oversubscription) vs. when it is run individually on the head node.
This might totally not be related to the RDKit at all, and we are currently approaching this issue from multiple angles.

Q: Is there a way to force the RDKit to run single-threaded? Or is that the default anyway?

Many thanks.

KR Axel

**Comment by bp-kelley** [comment]
Some functions are multithreaded, but the majority are not.  Look for numThreads and/or nthreads in the architecture.  Thinks like conformed embedding, substructlibrary are threaded for instance.

if you need to build a single threaded version we can show you how.

**Reply by apahl** [reply]
Thank you. So there is no way to force single-threading from the outside, like with an environment variable?

**Reply by greglandrum** [reply, marked as answer]
Unless we made a mistake, none of the code should use threading unless you specifically ask for it.

If the functions you are using do not have an argument (or parameter) related to the number of threads, then they should not be running multi-threaded

**Reply by apahl** [reply]
Ok, thank you.
